cpp/core/convert_core_json_to_marc.cc

```diff
@@ -0,0 +1,386 @@
+{
+    unsigned skipped_dupe_count(0), generated_count(0), skipped_incomplete_count(0);
+    for (auto &entry : root_array) {
+        const auto entry_object(JSON::JSONNode::CastToObjectNodeOrDie("entry", entry));
+        const auto id(std::to_string(entry_object->getIntegerValue("id")));
+        const auto control_number("CORE" + id);
+        if (ignore_unique_id_dups and unique_id_to_date_map->keyIsPresent(control_number))
+            ++skipped_dupe_count;
+        else {
+            MARC::Record new_record(MARC::Record::TypeOfRecord::LANGUAGE_MATERIAL, MARC::Record::BibliographicLevel::MONOGRAPH_OR_ITEM,
+                                    control_number);
+            std::set<std::string> authors;
+            if (not ProcessAuthors(*entry_object, &new_record, &authors)) {
+                ++skipped_incomplete_count;
+                continue;
+            }
+            ProcessContributors(*entry_object, &new_record, authors);
+            if (not ProcessTitle(*entry_object, &new_record)) {
+                ++skipped_incomplete_count;
+                continue;
+            }
+            new_record.insertControlField("007", "cr||||");
+            new_record.insertField("035", 'a', "(core)" + id);
+            new_record.insertField("591", 'a', "Metadaten maschinell erstellt (TUKRIM)");
+            new_record.insertField("852", 'a', project_sigil);
+            ProcessYear(*entry_object, &new_record);
+            ProcessDownloadURL(*entry_object, &new_record);
+            ProcessLanguage(*entry_object, &new_record);
+            ProcessAbstract(*entry_object, &new_record);
+            ProcessUncontrolledIndexTerms(*entry_object, &new_record);
+            ProcessYearPublished(*entry_object, &new_record);
+            ProcessJournal(*entry_object, issns_to_journal_titles_ppns_and_issns_map, unmatched_issns_to_counts_map, &new_record);
+            Process935Entries(_935_entries, &new_record);
+            marc_writer->write(new_record);
+            unique_id_to_date_map->addOrReplace(control_number, TimeUtil::GetCurrentDateAndTime());
+            ++generated_count;
+        }
+    }
+
+    std::cout << "Skipped " << skipped_dupe_count << " dupes and " << skipped_incomplete_count
+              << " incomplete entry/entries and generated " << generated_count << " MARC record(s).\n";
+}
+
+
+void GenerateUnmappedISSNList(File * const unmatched_issns_file,
+                              const std::unordered_map<std::string, unsigned> &unmatched_issns_to_counts_map)
+{
+    for (const auto &issn_and_count : unmatched_issns_to_counts_map)
+        (*unmatched_issns_file) << issn_and_count.first << '\t' << issn_and_count.second << '\n';
+    LOG_INFO("Wrote a list of " + std::to_string(unmatched_issns_to_counts_map.size()) + " unmapped ISSN's to \""
+             + unmatched_issns_file->getPath() + "\".");
+}
+
+
+const std::string UNIQUE_ID_TO_DATE_MAP_PATH(UBTools::GetTuelibPath() + "convert_json_to_marc.db");
```

Comment:
This should also be renamed to convert_core_json_to_marc.db

Reply:
Done!